### ISSUE TITLE
If expressions in If expression

### ISSUE THREAD

**Issue by hexaredecimal** [hexaredecimal/ML#5]
Currently the compiler emits invalid Java  code when another if expression is used in the body of the if expression. So the following code passes on the compiler but generates the wrong code.

```ml
if true then {
   1
} else (* body of the else *)
   if true then { 2 } else { 3 }
```

**Comment by hexaredecimal** [comment]
A solution that I have in mind is to keep track of how deep are we with if statements because it ALWAYS has else. if the count  > root then generate expressions and store the results in local variables. Unwind the recursion and emit a unary checking and returning different expressions and the locals created. This is one solution that I'm going to implement right now.

**Comment by hexaredecimal** [comment]
Fixed in commit 07e6cd4. It was very simple to implement, I guess I was just lazy for the past months.
